### PR TITLE
fix(ripple): Trigger on key events

### LINKED ISSUE #41: Ripple doesn't trigger on key down event
Steps to reproduce:

1. Start screenshot tests
1. Go to button screenshots
1. Bring to focus a button using the keyboard
1. While focused on a button, press space bar or enter key

Actual Results:
No Ripple is activated

Expected Results:
Ripple should activate, similar to MDC Web

### PR DESCRIPTION
**Pull request by bonniezhou**
Fixes #41.

Re-implement the `isSurfaceActive()` adapter method to check for active state directly on the ripple surface element, instead of creating and updating an `isActivated` state variable. 
Wire up the appropriate event handlers to call `deactivate(e)` in the foundation.

### CODE REVIEW

File: packages/ripple/index.js [hunk 1]
```diff
@@ -34,10 +34,14 @@ const withRipple = (WrappedComponent) => {
     }
 
     createAdapter_ = (instance) => {
+      const MATCHES = [
```
Comment (moog16) [inline packages/ripple/index.js]: can we import the ./utils method instead of copy/pasting here

File: test/unit/ripple/index.test.js [hunk 2]
```diff
@@ -85,6 +93,24 @@ test('touchStart event triggers activateRipple with no onTouchStart handler', ()
...
+  const touchEndHandler = td.func();
+  const wrapper = mount(<DivRipple onTouchEnd={touchEndHandler}/>);
+  const foundation = wrapper.instance().foundation_;
+  foundation.deactivate = td.func();
+  wrapper.simulate('touchEnd');
+  td.verify(foundation.deactivate(td.matchers.isA(Object)), {times: 1});
+  td.verify(touchEndHandler(td.matchers.isA(Object)), {times: 1});
+});
+
+test('touchEnd event triggers deactivateRipple with no onTouchEnd handler', () => {
```
Comment (moog16) [inline test/unit/ripple/index.test.js]: does this test do the same thing as line 102? I think remove line 102 & line 100

File: test/unit/ripple/index.test.js [hunk 3]
```diff
@@ -109,27 +135,32 @@ test('keyDown event triggers activateRipple with no onKeyDown handler', () => {
...
+  const wrapper = mount(<DivRipple onKeyUp={keyUpHandler}/>);
+  const foundation = wrapper.instance().foundation_;
+  foundation.deactivate = td.func();
+  wrapper.simulate('keyUp');
+  td.verify(foundation.deactivate(td.matchers.isA(Object)), {times: 1});
+  td.verify(keyUpHandler(td.matchers.isA(Object)), {times: 1});
 });
 
-test('#adapter.isUnbounded returns false prop is not set', () => {
+test('keyUp event triggers deactivateRipple with no onKeyUp handler', () => {
```
Comment (moog16) [inline test/unit/ripple/index.test.js]: same with this test, you probably don't need to test it in the above test

**Comment by kfranqueiro** [comment]
The enter key typically does not cause `:active` to be applied. Only space does. Test it with a native button: https://codepen.io/kfranqueiro/pen/KRjGZO

This should ostensibly already be handled by the MDC Web ripple foundation, so I'm not sure why it needs to be reimplmented in React... https://github.com/material-components/material-components-web/blob/v0.35.2/packages/mdc-ripple/foundation.js#L340

**Comment by moog16** [comment]
Gotcha...I thought space and enter keys were supposed to do the same with buttons, etc. 

RE Events: It is handled by the foundation, but we chose to not use the native html events, but instead use React synthetic events. The result of this is having to setup those event in the React Component.

File: test/unit/ripple/index.test.js (hunk 2 above)
Comment (bonniezhou) [reply test/unit/ripple/index.test.js]: These are two different tests, one to check that the ripple deactivates when there's no handler and one for when there is. The `DivRipple` that's mounted in each of these tests are different.

**Pull request merged**